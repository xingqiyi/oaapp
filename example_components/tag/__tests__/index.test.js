import React from 'react';
import { render, shallow } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Tag from '../index';

describe('SegmentedControl', () => {
  it('renders correctly', () => {
    const wrapper = render(<Tag>Basic</Tag>);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('small size does not have closeDom', () => {
    const wrapper = shallow(<Tag small closable>Basic</Tag>);
    expect(wrapper.find('.am-tag-close')).toHaveLength(0);
  });

  it('onChange then selected', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Tag onChange={onChange}>Basic</Tag>);
    expect(wrapper.state('selected')).toEqual(false);
    wrapper.find('.am-tag').simulate('click');
    expect(wrapper.state('selected')).toEqual(true);
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('onClose and removed', () => {
    const onClose = jest.fn();
    const afterClose = jest.fn();
    const wrapper = shallow(<Tag closable onClose={onClose} afterClose={afterClose}>Basic</Tag>);
    expect(wrapper.find('.am-tag-close')).toHaveLength(1);
    expect(wrapper.state('closed')).toEqual(false);
    wrapper.find('.am-tag-close').simulate('click');
    expect(onClose).toHaveBeenCalled();
    expect(wrapper.state('closed')).toEqual(true);
    expect(afterClose).toHaveBeenCalled();
    expect(wrapper.find('.am-tag')).toHaveLength(0);
  });
});
