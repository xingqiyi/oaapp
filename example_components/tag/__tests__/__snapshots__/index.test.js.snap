// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl renders correctly 1`] = `
<div
  class="am-tag am-tag-normal"
>
  <div
    class="am-tag-text"
  >
    Basic
  </div>
</div>
`;
